services/QuillLMS/app/models/classroom.rb

```diff
@@ -206,12 +217,15 @@ def google_classroom?
   def classroom_provider
     return 'Google Classroom' if google_classroom?
     return 'Clever' if clever_classroom?
+    return 'Canvas' if canvas_classroom?
   end
 
   def classroom_external_id
     return google_classroom_id if google_classroom?
 
     return clever_id if clever_classroom?
+
+    return canvas_classroom&.classroom_external_id
   end
```

Comment:
With only three options, this probably isn't a big deal, but I wonder if in the long term we should look into some sort of lookup-Hash for types rather than a bunch of `if` statements.  Maybe not worth worrying about if this is the only function that does this, but I often prefer hashes like that over a whole bunch of `if` clauses.  I could imagine something like:
```ruby
def classroom_external_id
  {
    'Google' => -> (classroom) { classroom.google_classroom_id },
    'Clever' => -> (classroom) { classroom.clever_id },
    'Canvas' => -> (classroom) { classroom.canvas_classroom&.classroom_external_id }
  }.fetch(classroom_provider).call(self)
end
```

Reply:
I think this is an interesting pattern.  I  will make a note to consider it when I refactor out clever_id and google_classroom_id from the classroom table.